src/metabase/lib/schema/filter.cljc

```diff
@@ -0,0 +1,17 @@
+(ns metabase.lib.schema.filter
+  "Schemas for the various types of filter clauses that you'd pass to `:filter` or use inside something else that takes
+  a boolean expression."
+  (:require
+   [metabase.lib.schema.common :as common]
+   [metabase.util.malli.registry :as mr]))
+
+(mr/def ::=
+  [:catn
+   [:clause [:= :=]]
+   [:options ::common/options]
+   ;; avoid circular refs between these namespaces.
```

Comment:
I think you can use clojure 1.11.1's new `:as-alias` to defeat this.
`[metabase.lib.schema.expression :as-alias expression]`